src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -56,6 +56,80 @@
                                        ["code", "lib", "func", "reason"])
 
 
+def _encode_asn1_int(backend, x):
+    i = backend._lib.ASN1_INTEGER_new()
+    # i = backend._ffi.gc(i, backend._lib.ASN1_INTEGER_free)
+    backend._lib.ASN1_INTEGER_set(i, x)
+    return i
+
+
+def _encode_asn1_str(backend, x, n):
+    s = backend._lib.ASN1_OCTET_STRING_new()
+    # s = backend._ffi.gc(s, backend._lib.ASN1_OCTET_STRING_free)
```

Comment:
Took a look at the source code for this one.  It seems like `X509_EXTENSION_create_by_OBJ()` passes the data to `X509_EXTENSION_set_data()` which creates a copy of the data without taking ownership.  I uncommented the GC line and everything passes without crashing, so I'll submit it with the GC statement to avoid a leak.